MatrixSDK/MXRestClient.m

```diff
@@ -318,6 +321,9 @@ - (MXHTTPOperation*)registerOrLoginWithUser:(MXAuthAction)authAction user:(NSStr
                              // Workaround: HS does not return the right URL. Use the one we used to make the request
```

Comment:
This is not actually a bug, according to the spec (https://matrix.org/docs/spec/#login) the key 'home_server' provides the hostname of the Home Server.

I think this key was previously used to return the home server url. So matrixSDK defined the field 'homeServer' to store the server url and not the hostname. This was the origin of this workaround.

I don't think synapse is able to rename this key ('home_server') in server response, so
we should refactor this field in matrixSDK. I just created SYIOS-155.